src/app/api/utils/auth.ts

```diff
@@ -222,7 +226,43 @@ export const authOptions: AuthOptions = {
       }
       if (token.subscriber) {
         session.user.subscriber = token.subscriber;
+
+        // refresh token
+        const dbFxATokens = await getFxATokens(token.subscriber.id);
+        if (
+          !dbFxATokens.fxa_session_expiry ||
+          dbFxATokens.fxa_session_expiry.getTime() < Date.now()
+        ) {
```

Comment:
I think the `getFxATokens()` can return `null` which might explode these if we're trying to access props on a null object.

Reply:
Ah great eye, and that's another advantage of using `.first()` instead of `.select()` in that query - then TS would've warned us of this.

Reply:
I simplified it a little bit so it uses first() and null check.

`first()` won't throw an error, it would just return `undefined` if it's not available